src/sentry/models/widget.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import absolute_import, print_function
+
+from django.db import models
+from django.utils import timezone
+from jsonfield import JSONField
+
+from sentry.constants import ObjectStatus
+from sentry.db.models import BoundedPositiveIntegerField, FlexibleForeignKey, Model, sane_repr
+
+
+class WidgetDisplayTypes(object):
+    LINE_CHART = 0
+    AREA_CHART = 1
+    STACKED_AREA_CHART = 2
+    BAR_CHART = 3
+    PIE_CHART = 4
+    TABLE = 5
+    WORLD_MAP = 6
+    PERCENTAGE_AREA_CHART = 7
+
+    @classmethod
+    def as_choices(cls):
+        return [
+            (cls.LINE_CHART, 'line'),
+            (cls.AREA_CHART, 'area'),
+            (cls.STACKED_AREA_CHART, 'stacked_area'),
+            (cls.BAR_CHART, 'bar'),
+            (cls.PIE_CHART, 'pie'),
+            (cls.TABLE, 'table'),
+            (cls.WORLD_MAP, 'world_map'),
+            (cls.PERCENTAGE_AREA_CHART, 'percentage_area_chart')
+        ]
+
+
+class WidgetDataSourceTypes(object):
+    DISCOVER_SAVED_SEARCH = 0
+    API_QUERY = 1  # ehhhh.....
+
+    @classmethod
+    def as_choices(cls):
+        return [
+            (cls.DISCOVER_SAVED_SEARCH, 'discover_saved_search'),
+            (cls.API_QUERY, 'api_query')
+        ]
+
+
+class WidgetDataSource(Model):
+    """
+    A dashboard widget.
+    """
+    __core__ = True
+
+    widget = FlexibleForeignKey('sentry.Widget')
+    type = BoundedPositiveIntegerField(choices=WidgetDataSourceTypes.as_choices())
+    name = models.CharField(max_length=128)
+    data = JSONField(default={})
```

Comment:
What are we going to store in here?

Reply:
A discover query, this is the as the discover saved query model: https://github.com/getsentry/sentry/blob/master/src/sentry/models/discoversavedquery.py#L31

Reply:
It's possible in the future that a widget data source can be a sentry API - which would then be a simple string and we'd handle mapping that to an endpoint in the frontend.